components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java

```diff
@@ -204,6 +189,13 @@ public void run() {
                 payload.put(APIConstants.API_SPEC_NAME, api.getId().getApiName());
                 payload.put(APIConstants.TENANT_DOMAIN, apiEvent.getTenantDomain());
                 payload.put(APIConstants.VERSION, apiEvent.getApiVersion());
+                payload.put(APIConstants.VISIBILITY, apiEvent.getApiVisibility());
```

Comment:
Can we not provide the role based access control using only VISIBILITYROLES? What is the need to have this VISIBILITY value?

Reply:
Currently we are only using VISIBILITYROLES for filtering.

Reply:
We removed VISIBILITY